8/debian-10/rootfs/opt/bitnami/scripts/grafana/entrypoint.sh

```diff
@@ -14,13 +14,37 @@ set -o pipefail
 . /opt/bitnami/scripts/libbitnami.sh
 . /opt/bitnami/scripts/liblog.sh
 
+function is_exec() {
+    # This checks if the first provided argument is executable or if only args was used
+    exec_in_path=$(builtin type -P "$1" 2> /dev/null)
+    if [[ -f "$1" && -x $(realpath "$1" 2> /dev/null) || -x $(realpath "$exec_in_path" 2> /dev/null) ]]; then
+        true;
+    else
+        false;
+    fi;
+}
+
+
 print_welcome_page
 
 if [[ "$1" = "/opt/bitnami/scripts/grafana/run.sh" ]]; then
-    /opt/bitnami/scripts/grafana/setup.sh
+    # This catches the error-code from libgrafana.sh for the immediate exit when the grafana-operator is used. And ensure that the exit code is kept silently.
+    /opt/bitnami/scripts/grafana/setup.sh && GRAFANA_OPERATOR_IMMEDIATE_EXIT=$? || GRAFANA_OPERATOR_IMMEDIATE_EXIT=$?
+    if [ $GRAFANA_OPERATOR_IMMEDIATE_EXIT -ne 0 ] && [ $GRAFANA_OPERATOR_IMMEDIATE_EXIT -ne 255 ]; then
+        exit $GRAFANA_OPERATOR_IMMEDIATE_EXIT
+    fi
     /post-init.sh
     info "** Grafana setup finished! **"
+    if [ $GRAFANA_OPERATOR_IMMEDIATE_EXIT -eq 255 ]; then
```

Comment:
For this, I don't understand. Why is this logic here if we already check it above? Unless Grafana Operator makes explicit use of `post-init.sh`, I don't see why it would be required.

Also:

```suggestion
    if [[ "${GRAFANA_OPERATOR_IMMEDIATE_EXIT:-0}" -eq 255 ]; then
```